test/unit/index.js

```diff
@@ -48,6 +41,55 @@ describe("asyncPool", function() {
         await asyncPool(2, [100, 500, 300, 200], timeout);
         expect(results).to.deep.equal([100, 300, 500, 200]);
       });
+
+      it("runs all promises in parallel when the pool is bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise(resolve =>
+            setTimeout(() => {
+              results.push(i);
+              resolve();
+            }, i)
+          );
+        await asyncPool(5, [100, 500, 300, 200], timeout);
+        expect(results).to.deep.equal([100, 200, 300, 500]);
+      });
+
+      it("rejects on error with pool limit effective", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise((resolve, reject) =>
+            setTimeout(() => {
+              results.push(i);
+
+              if (i === 300) {
+                reject(new Error("Oops"));
+              } else {
+                resolve();
+              }
+            }, i)
+          );
+        await rejects(asyncPool(2, [100, 500, 300, 200], timeout));
```

Comment:
Can we write this test using `chai` instead of using `rejects` from `assert`? Therefore:

```diff
- await rejects(asyncPool(2, [100, 500, 300, 200], timeout));
+ return expect(asyncPool(2, [100, 500, 300, 200], timeout)).to.be.rejected
```

https://www.chaijs.com/plugins/chai-as-promised/
https://www.chaijs.com/api/bdd/#method_throw

Reply:
Why? The existing test code using `rejects` actually never produces failing test.

Using chai, we get the below output in case this test fails, e.g., using array `[100, 100, 100]`.
```
  1) asyncPool
       ES6 support
         rejects on error with pool limit effective:
     AssertionError: expected promise to be rejected but it was fulfilled with [ Array(4) ]
```

Reply:
Actually, I tried the chai version first but couldn't make the test fail - probably because I missed to `return` the result. With `rejects` it did work for me (i.e. I could make the test fail), but since your proposal works as well I am going to use it.

Reply:
:+1: thanks